### PR TITLE
Report lazyload timings and send to beacon

### PR DESCRIPTION
**Pull request by aintgoin2goa**
@ironsidevsquincy @wheresrhys 

I think this will work, but not 100% sure if I've done the right thing

### CODE REVIEW

File: src/lib/lazy-load.js [hunk 1]
```diff
@@ -1,18 +1,72 @@
...
+const uid = () => (Date.now() * Math.random()).toString(16);
+
+const setUid = (img) => {
+	const id = uid();
+	img.setAttribute('data-uid', id);
+	return id;
+};
+
+const getUid = (img) => {
+	return img.dataset && img.dataset.uid ? img.dataset.uid : img.getAttribute('data-uid');
```
Comment (ironsidevsquincy) [inline src/lib/lazy-load.js]: unfortunately `dataset` isn't polyfilled

File: src/lib/lazy-load.js [hunk 2]
```diff
@@ -1,18 +1,72 @@
 const lazyLoadingModifier = 'lazy-loading';
 const lazyLoadingImageClass = `n-image--${lazyLoadingModifier}`;
 const lazyLoadingWrapperClass = `n-image-wrapper--${lazyLoadingModifier}`;
+const performanceSupported = ('performance' in window && typeof window.performance.mark === 'function');
+
+const uid = () => (Date.now() * Math.random()).toString(16);
```
Comment (ironsidevsquincy) [inline src/lib/lazy-load.js]: should all these be `uuid` (rather than `uid`)?

File: src/lib/lazy-load.js (hunk 1 above)
Comment (ironsidevsquincy) [inline src/lib/lazy-load.js]: Oh, I see. Why not just `return img.getAttribute('data-uid');`?

File: src/lib/lazy-load.js [hunk 3]
```diff
@@ -1,18 +1,72 @@
 const lazyLoadingModifier = 'lazy-loading';
 const lazyLoadingImageClass = `n-image--${lazyLoadingModifier}`;
 const lazyLoadingWrapperClass = `n-image-wrapper--${lazyLoadingModifier}`;
+const performanceSupported = ('performance' in window && typeof window.performance.mark === 'function');
```
Comment (wheresrhys) [inline src/lib/lazy-load.js]: `require('n-ui/utils').perfMark` does all this sort of stuff https://github.com/Financial-Times/n-ui/blob/master/utils/index.js#L59. Also hooks up to LUX -the new tool we're trialling from speedcurve

File: src/lib/lazy-load.js (hunk 2 above)
Comment (aintgoin2goa) [inline src/lib/lazy-load.js]: Dunno - what's the different?

File: src/lib/lazy-load.js (hunk 3 above)
Comment (aintgoin2goa) [inline src/lib/lazy-load.js]: Yeah but how do I use it from here?  Spent hours trying to figure that out before giving up (also why I'm not using `broadcast` from utils

File: src/lib/lazy-load.js (hunk 2 above)
Comment (ironsidevsquincy) [inline src/lib/lazy-load.js]: guess it depends how big the universe is. but yeah, fair enough

File: src/lib/lazy-load.js (hunk 1 above)
Comment (ironsidevsquincy) [inline src/lib/lazy-load.js]: why only chrome? performance marks work in most browsers (chrome, IE, etc)

Comment (aintgoin2goa) [inline src/lib/lazy-load.js]: Video issue reported as chrome-specific.  Is there a browser that has `peformance` but not `dataset`?

Comment (ironsidevsquincy) [inline src/lib/lazy-load.js]: e.g. IE01. Basically questioning why not go for the simpler `return img.getAttribute('data-uid');`? Works in all browsers, no need to worry about browser support

**Pull request merged**